sync:
  image: n3llyb0y/wait
  links:
    - auth
    - ari
    - confd
    - ctid-ng
    - rabbitmq
  environment:
    PORTS: "5039 5672 9486 9497 9500"
    TIMEOUT: "${INTEGRATION_TEST_TIMEOUT}"

ari:
  image: ari-mock
  ports:
    - "5039"
  volumes:
    - "./ari:/usr/local/share/ari"
    - "./ssl/ari:/usr/local/share/ari-ssl"
  command: "python /usr/local/share/ari/mock_ari_fail.py 5039"

auth:
  image: wazopbx/xivo-auth-mock
  expose:
    - "9497"
  volumes:
    - "./ssl:/usr/local/share/ssl"

confd:
  image: p0bailey/docker-flask
  ports:
    - "9486"
  volumes:
    - "./xivo-confd:/usr/local/share/xivo-confd"
    - "./ssl:/usr/local/share/ssl"
  command: "python /usr/local/share/xivo-confd/mock-xivo-confd.py 9486"

ctid-ng:
  image: xivo-ctid-ng-test
  volumes:
    - "../../..:/usr/src/xivo-ctid-ng"
    - "./etc/xivo-ctid-ng:/etc/xivo-ctid-ng"
    - "./ssl:/usr/local/share/ssl"
  ports:
    - "9500"
  environment:
    XIVO_UUID: "08c56466-8f29-45c7-9856-92bf1ba89b92"
  links:
    - ari
    - auth
    - confd
    - rabbitmq

rabbitmq:
  image: rabbitmq
  ports:
    - "5672"
